extensions/vertx-http/deployment/src/main/java/io/quarkus/devui/deployment/menu/WorkspaceProcessor.java

```diff
@@ -74,7 +75,8 @@ void locateWorkspaceItems(BuildSystemTargetBuildItem buildSystemTarget,
                 Files.walkFileTree(projectRoot, new SimpleFileVisitor<Path>() {
                     @Override
                     public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs) throws IOException {
-                        if (Files.isHidden(dir) || ignoreFolders.contains(dir.getFileName().toString())) {
+                        if (Files.isHidden(dir) || ignoreFolders.contains(dir.getFileName().toString())
+                                || !Files.isReadable(dir)) {
```

Comment:
For a directory, you also need executable. At least on Unix. Not sure how Windows reacts when checking for executable...

Reply:
Thanks, maybe that's why I had to add `visitFileFailed` (although I'll probably keep it anyway). I'll try it.

Reply:
Yes keep it, it’s nice to have a safe guard.

Reply:
I have added a check for "executable" file permission and it is in the PR now. But when I tried to reproduce it without `visitFileFailed`, application still failed to start. So I'll keep it as you are OK with it.